src/pages/index.js

```diff
@@ -62,17 +115,104 @@ cardAddBtn.addEventListener("click", () => {
   newCardPopup.open();
 });
 
+pfp.addEventListener("click", () => {
+  EditPfp.open();
+});
+
 //Handler
+function handleDeleteCard(card) {
+  confirmPopup.open();
+  confirmPopup.handleConfirm(() => {
+    confirmPopup.showLoading(true);
+    api
+      .deleteCard(card.getCardId())
+      .then(() => {
+        card.handleDeleteCard();
+        confirmPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        confirmPopup.showLoading(false);
+      });
+  });
+}
+
 function editProfileSubmit(userData) {
+  profilePopup.showLoading(true);
+  api
+    .editUserData({
+      name: userData.name,
+      description: userData.description,
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profilePopup.showLoading(false);
+    });
   userInfo.setUserInfo(userData);
   profilePopup.close();
```

Comment:
You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error, it shouldn’t change